### PR TITLE
Fix bugs for 1.0.1

### PR DESCRIPTION
**Pull request by OsvaldoRosado**
Address #352, #351, #349, #348 and prepare a 1.0.1 release.

### CODE REVIEW

File: AutoCollection/HttpDependencies.ts [hunk 1]
```diff
@@ -55,29 +56,43 @@ class AutoCollectHttpDependencies {
...
-                const request: http.ClientRequest = originalHttpsRequest.call(
-                    https, options, ...requestArgs);
-                if (request && options && !(<any>options)[AutoCollectHttpDependencies.disableCollectionRequestOption]) {
-                    AutoCollectHttpDependencies.trackRequest(this._client, options, request);
-                }
-                return request;
-            };
-        }
+        https.request = (options, ...requestArgs: any[]) => {
+            const request: http.ClientRequest = originalHttpsRequest.call(
```
Comment (KamilSzostak) [inline AutoCollection/HttpDependencies.ts]: isn't it exactly the same as for http.request? Would it make sense to move it to a method and just pass originalHttpsRequest/originalRequest object?

Comment (OsvaldoRosado) [reply AutoCollection/HttpDependencies.ts]: Nope, the base functions take different numbers of arguments 😄

Comment (OsvaldoRosado) [reply AutoCollection/HttpDependencies.ts]: I do really dislike the duplication though..

Comment (KamilSzostak) [reply AutoCollection/HttpDependencies.ts]: Ok, let's leave it as it is.

Comment (OsvaldoRosado) [reply AutoCollection/HttpDependencies.ts]: Actually scratch that, I think this can be cleanly de-duplicated. Will change

**Pull request merged**